src/main/java/com/blackducksoftware/ohcount4j/SourceFileUtils.java

```diff
@@ -45,21 +45,32 @@ public static String head(SourceFile sourceFile) throws IOException {
             try (SourceFile srcFile = new SourceFile(sourceFile.getPath())) {
                 return srcFile.head(HEAD_SIZE);
             }
-        } else if (sourceFile.getContents() != null) {
-            // the source is not from the file, so we got to have content initialized
-            return new String(sourceFile.getContents());
         } else {
-            return null;
+        	char contents[] = sourceFile.getContents() != null ? sourceFile.getContents() : new char[0];
+        	// the source is not from the file, so we got to have content initialized
+        	return new String(contents);
```

Comment:
This looks nearly identical to the code in the `AbstractExtnResolver.getContents` method in #15.

This seems to be a more logical place for this logic, this being a Source File Utility. Could this remove the need for that other code in `AbstractExtnResolver.getContents`?